mountpoint-s3-fs/src/superblock/readdir.rs

```diff
@@ -193,12 +195,16 @@ impl ReaddirHandle {
                     *restore_status,
                     inner.config.cache_config.file_ttl,
                 );
-                Some(RemoteLookup {
+                RemoteLookup {
                     stat,
                     kind: InodeKind::File,
-                })
+                }
             }
-        }
+        };
+
+        // Remote entry, update the superblock.
+        let lookup = inner.update_from_remote(self.dir_ino, name, Some(remote_lookup))?;
+        Ok(Some(lookup))
```

Comment:
Could be simplified. E.g.:
```suggestion
        inner.update_from_remote(self.dir_ino, name, Some(remote_lookup))
```